Modules/Filtering/FFT/include/itkVnlFFTImageFilterFactory.h

```diff
@@ -0,0 +1,70 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkVnlFFTImageFilterFactory_h
+#define itkVnlFFTImageFilterFactory_h
+#include "ITKFFTExport.h"
+
+#include "itkObjectFactoryBase.h"
+
+namespace itk
+{
+/**
+ *\class VnlFFTImageFilterFactory
+ * \brief Create instances of Vnl FFT ImageFilter objects using an object factory.
+ * \ingroup ITKFFT
+ */
+class ITKFFT_EXPORT VnlFFTImageFilterFactory : public ObjectFactoryBase
```

Comment:
@thewtex The heart of what I'm trying, and maybe failing, to communicate is that the current implementations of `VnlFFTImageFilterFactory` and `FFTWFFTImageFilterFactory` could inherit from `LightObject` without any inherent loss of meaning. They exist solely to initialize FFT factories and nothing more. Sort of like [`itkCenteredTransformInitializer`](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Registration/Common/include/itkCenteredTransformInitializer.h) exists solely to initialize a transform, but is not itself a transform.

Our CMake architecture does not accommodate factory initializers, thus under existing architecture these must be named ending in `FFTImageFilterFactory` as opposed to `FFTImageFilterFactoryInitializer`.

Reply:
Yes, we should make changes to these classes.

The existing factory architecture does have a constraint on `Factory$`. And we may not be able to use `FFTImageFilterFactoryInitializer` exactly as initially proposed. However, `Initializer` here is the same as `CenteredTransformInitializer` so a different term may be helpful conceptually, too. What do you prefer?

- `FFTInitFactory`
- `FFTImageFilterInitFactory`
- `FFTInitializeFactory`
- something else?

Reply:
If we are stuck with the `Factory$` naming then any of the above seem equally reasonable to me. Under the architecture you're envisioning are there any remaining prefix constraints, i.e. `FFTImageFilter` has to be in/excluded, etc?

Reply:
We should be able to use any prefix.